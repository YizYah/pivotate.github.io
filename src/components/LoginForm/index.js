import React, { useState } from "react";
import styled from "styled-components";
import Modal from "../Modal";
import { withNoStack } from "@nostack/no-stack";

import ForgotPasswordButton from "../ForgotPasswordButton";

const LoginForm = (props, { loading, currentUser, login }) => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [isSubmitting, setIsSubmitting] = useState(false);
  const [error, setError] = useState("");

  if (loading || currentUser) {
    return null;
  }

  const handleSubmit = async e => {
    e.preventDefault();

    setIsSubmitting(true);

    try {
      await login({
        username,
        password
      });
    } catch (error) {
      setError(
        error.message ||
          (error.graphQLErrors &&
            error.graphQLErrors.length &&
            error.graphQLErrors[0]) ||
          error
      );
      setIsSubmitting(false);
    }
  };

  return (
    <Modal onClose={props.onClose}>
      <div className="dialog__title">
        <img
          src="http://pivotatestaticassets.com/images/Pivotate Logo.svg"
          width="170"
        />
        <h3>LOGIN</h3>
      </div>
      <form onSubmit={handleSubmit} className="form">
        <div className="form__input">
          <label htmlFor="nostack-username">
            <input
              id="nostack-username"
              type="text"
              name="username"
              placeholder="Username"
              disabled={isSubmitting}
              value={username}
              onChange={e => setUsername(e.target.value)}
            />
          </label>
        </div>
        <div className="form__input">
          <label htmlFor="nostack-password">
            <input
              id="nostack-password"
              type="password"
              placeholder="Password"
              name="password"
              disabled={isSubmitting}
              value={password}
              onChange={e => setPassword(e.target.value)}
            />
          </label>
        </div>
        <div className="form__input">
          <button
            className="button button--yellow"
            type="submit"
            disabled={isSubmitting || !username || !password}
          >
            Log In
          </button>
        </div>
        {error && <div className="form__input">{error}</div>}

        <br />
        {/* <div>
          <small>or</small>
        </div> */}
        {/* Temporarility Hide Social icons */}
        {/* <div className="flex justify-space-between">
          <button type="button" className="button button--fb">
            Log in with
            <img className="social-icon" height="14" src="http://pivotatestaticassets.com/images/facebook.png" />
          </button>
          <div style={{ width: "50px" }}></div>
          <button type="button" className="button button--google">
            Log in with
            <img className="social-icon" height="14" src="http://pivotatestaticassets.com/images/google plus.png" />
          </button>
        </div> */}
        <br />
        <small>
          Dont have an account?
          <a
            href="javascript:void(0);"
            className="teal--text"
            onClick={props.onSwitch}
          >
            &nbsp; Sign Up
          </a>
        </small>
      </form>
      {/* <div className="form__input">
        <ForgotPasswordButton />
      </div> */}
    </Modal>
  );
};

export default withNoStack(LoginForm);
