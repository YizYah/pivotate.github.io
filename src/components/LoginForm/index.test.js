import { shallow } from 'enzyme';
import 'whatwg-fetch';
import React from 'react';
import LoginForm from '.';
jest.mock('@nostack/no-stack', () => ({
  withNoStack: (el) => el,
}));

jest.mock('../ForgotPasswordButton', () => () => <input type='submit' />);
let login, loading, currentUser, open, onClose;

describe('LoginForm component', () => {
  describe('AND NO user provided and NOT loading', () => {
    beforeEach(() => {
      login = jest.fn();
      loading = false;
      currentUser = null;
      open = true;
      onClose = jest.fn();
    });
    it('SHOULD render login', () => {
      const output = shallow(
        <LoginForm
          login={login}
          loading={loading}
          currentUser={currentUser}
          open={open}
          onClose={onClose}
        />
      );

      expect(output).toMatchSnapshot();
    });

    it('SHOULD submit form', () => {
      const output = shallow(
        <LoginForm
          login={login}
          loading={loading}
          currentUser={currentUser}
          open={open}
          onClose={onClose}
        />
      );
    });
  });

  describe('AND  user is provided OR it is loading', () => {
    beforeEach(() => {
      login = jest.fn();
      loading = false;
      currentUser = {};
      open = true;
      onClose = jest.fn();
    });

    it('SHOULD NOT render login', () => {
      const output = shallow(
        <LoginForm
          login={login}
          loading={loading}
          currentUser={currentUser}
          open={open}
          onClose={onClose}
        />
      );

      expect(output).toMatchSnapshot();
    });
  });
});
