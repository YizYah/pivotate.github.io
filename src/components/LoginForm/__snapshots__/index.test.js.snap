// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component AND  user is provided OR it is loading SHOULD NOT render login 1`] = `""`;

exports[`LoginForm component AND NO user provided and NOT loading SHOULD render login 1`] = `
<TransitionsModal
  onClose={[MockFunction]}
  open={true}
>
  <div>
    <styled.div>
      <a
        href="/"
      >
        <img
          alt="Pivotate Logo"
          src="https://pivotatestaticassets.com/images/Pivotate Logo.svg"
          width="170"
        />
      </a>
    </styled.div>
    <WithStyles(ForwardRef(InputLabel))
      className="makeStyles-inputLabel-3"
    >
      Login
    </WithStyles(ForwardRef(InputLabel))>
  </div>
  <form
    onSubmit={[Function]}
  >
    <styled.div>
      <label
        htmlFor="nostack-username"
      >
        <WithStyles(ForwardRef(TextField))
          className="makeStyles-textField-1"
          disabled={false}
          id="nostack-username"
          label="Username"
          name="username"
          onChange={[Function]}
          type="text"
          value=""
          variant="outlined"
        />
      </label>
    </styled.div>
    <styled.div>
      <label
        htmlFor="nostack-password"
      >
        <WithStyles(ForwardRef(TextField))
          className="makeStyles-textField-1"
          disabled={false}
          id="nostack-password"
          label="Password"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
          variant="outlined"
        />
      </label>
    </styled.div>
    <styled.div>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-2"
        color="primary"
        disabled={true}
        type="submit"
        variant="contained"
      >
        Log In
      </WithStyles(ForwardRef(Button))>
    </styled.div>
  </form>
  <styled.div>
    <Component />
  </styled.div>
</TransitionsModal>
`;
